### ISSUE TITLE
Unplugging the Nitrokey results in an error (for certain systems)

### ISSUE THREAD

**Issue by Niklap97** [Nitrokey/nitrokey-app2#151]
Error Output:
```
Traceback (most recent call last):
  File "/home/test/dev/nitrokey-app2/venv/lib/python3.11/site-packages/nitrokeyapp/gui.py", line 149, in device_connect
    self.detect_removed_devices()
  File "/home/test/dev/nitrokey-app2/venv/lib/python3.11/site-packages/nitrokeyapp/gui.py", line 203, in detect_removed_devices
    nk3_list = [str(device.uuid())[:-4] for device in Nitrokey3Device.list()]
                                                      ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/test/dev/nitrokey-app2/venv/lib/python3.11/site-packages/pynitrokey/nk3/device.py", line 144, in list
    for device in CtapHidDevice.list_devices():
  File "/home/test/dev/nitrokey-app2/venv/lib/python3.11/site-packages/fido2/hid/__init__.py", line 261, in list_devices
    yield cls(d, open_connection(d))
                 ^^^^^^^^^^^^^^^^^^
  File "/home/test/dev/nitrokey-app2/venv/lib/python3.11/site-packages/fido2/hid/linux.py", line 51, in open_connection
    return LinuxCtapHidConnection(descriptor)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/test/dev/nitrokey-app2/venv/lib/python3.11/site-packages/fido2/hid/base.py", line 69, in __init__
    self.handle = 
os.open
(descriptor.path, os.O_RDWR)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/dev/hidraw10' 
```
- Could not be reconstructed with Ubuntu 22.04.3 LTS

**Comment by daringer** [comment]
looking for reproducer @jj-so

**Comment by daringer** [comment]
after the PR #173 the error (partly) still exists:
```
Traceback (most recent call last):
  File "/home/dariball/dev/nitrokey-app2/venv/lib/python3.11/site-packages/nitrokeyapp/gui.py", line 149, in device_connect
    self.detect_removed_devices()
  File "/home/dariball/dev/nitrokey-app2/venv/lib/python3.11/site-packages/nitrokeyapp/gui.py", line 204, in detect_removed_devices
    nk3_list = [str(device.uuid())[:-4] for device in Nitrokey3Device.list()]
                                                      ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dariball/dev/nitrokey-app2/venv/lib/python3.11/site-packages/pynitrokey/nk3/device.py", line 144, in list
    for device in CtapHidDevice.list_devices():
  File "/home/dariball/dev/nitrokey-app2/venv/lib/python3.11/site-packages/fido2/hid/__init__.py", line 261, in list_devices
    yield cls(d, open_connection(d))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dariball/dev/nitrokey-app2/venv/lib/python3.11/site-packages/fido2/hid/__init__.py", line 109, in __init__
    response = self.call(CTAPHID.INIT, nonce)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dariball/dev/nitrokey-app2/venv/lib/python3.11/site-packages/fido2/hid/__init__.py", line 176, in call
    self._connection.write_packet(packet.ljust(self._packet_size, b"\0"))
  File "/home/dariball/dev/nitrokey-app2/venv/lib/python3.11/site-packages/fido2/hid/linux.py", line 47, in write_packet
    super().write_packet(b"\0" + packet)
  File "/home/dariball/dev/nitrokey-app2/venv/lib/python3.11/site-packages/fido2/hid/base.py", line 76, in write_packet
    if os.write(self.handle, packet) != len(packet):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 19] No such device
```

**Comment by daringer** [comment]
fixed in v2.2.1
